annotations/s2i-annotations/src/main/java/io/dekorate/s2i/manifest/S2iManifestGenerator.java

```diff
@@ -91,9 +91,8 @@ public void generate(S2iBuildConfig config) {
       ImageConfiguration imageConfig = configurationRegistry
           .getImageConfig(BuildServiceFactories.supplierMatches(getProject()))
           .orElse(null);
-      String registry = Strings.isNullOrEmpty(imageConfig.getRegistry()) ? "docker.io" : imageConfig.getRegistry();
-
       if (imageConfig != null) {
+        String registry = Strings.isNullOrEmpty(imageConfig.getRegistry()) ? "docker.io" : imageConfig.getRegistry();
```

Comment:
This is to avoid null pointer exception which happens when we disable the s2i build.